app/ide-desktop/lib/dashboard/src/App.tsx

```diff
@@ -147,28 +148,32 @@ export interface AppProps {
  * This component handles all the initialization and rendering of the app, and manages the app's
  * routes. It also initializes an `AuthProvider` that will be used by the rest of the app. */
 export default function App(props: AppProps) {
-  const { supportsLocalBackend } = props
+  const { supportsLocalBackend, projectManagerUrl } = props
   // This is a React component even though it does not contain JSX.
   // eslint-disable-next-line no-restricted-syntax
   const Router = detect.isOnElectron() ? router.HashRouter : router.BrowserRouter
   const queryClient = React.useMemo(() => reactQueryClientModule.createReactQueryClient(), [])
-  const [rootDirectoryPath, setRootDirectoryPath] = React.useState<projectManager.Path | null>(null)
   const [error, setError] = React.useState<unknown>(null)
-  const isLoading = supportsLocalBackend && rootDirectoryPath == null
+  const [projectManager, setProjectManager] = React.useState<ProjectManager | null>(null)
```

Comment:
TLDR:
1) Create custom hook 'useProjectManager()`
2) configure `useQuery` to store the current project manager as long as needed (maybe forever?) By default, it stores data as long as it's in use and 5 minutes after unmount of the latest component that uses this data.
3) Use that hook in the app.
4) Additionally, we might want to use `useSuspenceQuery` to integrate fetching with `React.Suspense`
5) Also, there's a possibility to prefetch data with `usePrefetchQuery`, so it might be helpful to separate query options from `useQuery`, You can use `queryOptions` function for that: https://tanstack.com/query/latest/docs/framework/react/typescript#typing-query-options

Reply:
true, but as this is a fix i think it's better to avoid migrating to react-query for now, if we really *should* migrate then we can do that in another PR

Reply:
Not sure, because migrating to react query makes this PR obsolete(it'll be not the next step, but a whole rewrite, and it's relatively easy to rewrite). 

If you need any help with rewriting to `react-query` please reach me in the DM

Reply:
while that's true, the part i'm unsure about isn't so much about *using* react query...

... it's about (ab)using react query as a form of permanent global state. i think it'd make it harder for users to figure out how the project manager is being cached, especially if they are unfamiliar with React Query.

iow: we have all other globals in Contexts, i'd prefer staying consistent and *maybe* switch to React Query after figuring out what we should and shouldn't be keeping in React Query.

Reply:
I think it's okay to use React.Context to provide variables that don't require fetching or accessing asynchronously, such as clipboard or geolocation. For asynchronous tasks, I strongly recommend using react-query. There's a clear distinction between React.Context and react-query. As far as I can see, we need to fetch some data before we create a ProjectManager instance, which is why I suggest rewriting it using react-query.

Reply:
sure, but again the issue is not with using react-query, but rather with using react-query as hidden global state

Reply:
It's more about proper working with the async flow, integration with `React.Suspense`, prefetching, invalidation and so on.
The global state goes latter